pkg/frinkiac/http/quote.go

```diff
@@ -6,27 +6,62 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"strconv"
 	"strings"
 
-	"github.com/rs/zerolog/log"
 	"golang.org/x/net/html"
 )
 
-// QuoteResult represents a single result from a quote search
-type QuoteResult struct {
-	ImagePath string // e.g., /img/S09E22/202334/medium.jpg
-	Season    string // e.g., S09
-	Episode   string // e.g., E22
-	ID        string // e.g., 202334
-}
-
 // APISearchResult represents a single result from the Frinkiac API search endpoint
 type APISearchResult struct {
 	ID        int    `json:"Id"`
 	Episode   string `json:"Episode"` // e.g., S16E01
 	Timestamp int    `json:"Timestamp"`
 }
 
+// validateEpisodeFormat checks if the episode format is valid
+// Episode format is expected to be like "S16E01"
+func validateEpisodeFormat(episode string) error {
+	if len(episode) < 6 {
+		return fmt.Errorf("invalid episode format: %s", episode)
+	}
+	return nil
+}
+
+// GetSeasonAndEpisode extracts season and episode numbers from an APISearchResult
+// Episode format is expected to be like "S16E01"
+func GetSeasonAndEpisode(result APISearchResult) (season int, episode int, err error) {
+	if err := validateEpisodeFormat(result.Episode); err != nil {
+		return 0, 0, err
+	}
+
+	// Extract season number from "S16" part
+	seasonStr := result.Episode[1:3] // Skip 'S', get "16"
+	season, err = strconv.Atoi(seasonStr)
+	if err != nil {
+		return 0, 0, fmt.Errorf("invalid season number in episode %s: %w", result.Episode, err)
+	}
+
+	// Extract episode number from "E01" part
+	episodeStr := result.Episode[4:6] // Skip "S16E", get "01"
+	episode, err = strconv.Atoi(episodeStr)
+	if err != nil {
+		return 0, 0, fmt.Errorf("invalid episode number in episode %s: %w", result.Episode, err)
```

Comment:
Use. the introduced ErrEpisodeFormat.

Reply:
Introduce EpisodeID type def, use it in the result Field and rename the field to EpisodID